src/main/java/fi/istrange/traveler/auth/Authenticator.java

```diff
@@ -0,0 +1,47 @@
+package fi.istrange.traveler.auth;
+
+import fi.istrange.traveler.dao.CredentialDao;
+import fi.istrange.traveler.db.tables.pojos.UserCredentials;
+import org.jooq.DSLContext;
+
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.WebApplicationException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+/**
+ * Created by aleksandr on 18.4.2017.
+ */
+public class Authenticator {
+    private CredentialDao credentialDao;
+
+    public Authenticator() {
+        credentialDao = new CredentialDao();
+    }
+
+    public boolean authenticate(String username, String password, DSLContext database) {
+        if (username == null || password == null) {
+            throw new WebApplicationException(422);
+        }
+
+        String hashedPassword = password;
+        try {
+            hashedPassword = PasswordHasher.hashPassword(password);
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+
+
+        Optional<UserCredentials> credentials= credentialDao.fetchByUsername(username, database);
+
+        if (
+                !credentials.isPresent() ||
+                !hashedPassword.equals(credentials.get().getPassword()) ||
+                !credentials.get().getActive()
+        ) {
+            throw new NotAuthorizedException("Invalid credentials");
```

Comment:
Makes sense. I was thinking about keeping exceptions as they are, but yeah. I'll chnage it in a moment